std/gkr/testing.go

```diff
@@ -29,35 +30,13 @@ import (
 // This method only works under the test engine and should only be called to debug a GKR circuit, as the GKR prover's errors can be obscure.
 func (api *API) SolveInTestEngine(parentApi frontend.API) [][]frontend.Variable {
 	res := make([][]frontend.Variable, len(api.toStore.Circuit))
-	degreeTestedGates := make(map[string]struct{})
+	var degreeTestedGates sync.Map
```

Comment:
Why `sync.Map`? The variable is local to this method and it doesn't use it in parallel?